File: app/src/main/java/com/mss/imagesearcher/repositories/network/NetMapper.java
package com.mss.imagesearcher.repositories.network;

import androidx.annotation.NonNull;

import com.mss.imagesearcher.domain.models.ImageInfo;
import com.mss.imagesearcher.repositories.network.models.ImageInfoNet;

import java.util.ArrayList;
import java.util.List;

class NetMapper {

    @NonNull
    public static List<ImageInfo> mapFromNet(@NonNull final List<ImageInfoNet> imagesNet) {
        List<ImageInfo> images = new ArrayList<>();
        for (ImageInfoNet imageNet : imagesNet) {
            images.add(mapFromNet(imageNet));
        }
        return images;
    }

    @NonNull
    private static ImageInfo mapFromNet(@NonNull final ImageInfoNet imageInfoDB) {
        final ImageInfo imageInfo = new ImageInfo();
        imageInfo.setId(imageInfoDB.getId());
        imageInfo.setType(imageInfoDB.getType());
        imageInfo.setTags(imageInfoDB.getTags());
        imageInfo.setPageUrl(imageInfoDB.getPageUrl());
        imageInfo.setLargeImageURL(imageInfoDB.getLargeImageURL());
        imageInfo.setWebFormatURL(imageInfoDB.getWebFormatURL());
        imageInfo.setPreviewURL(imageInfoDB.getPreviewURL());
        imageInfo.setComments(imageInfoDB.getComments());
        imageInfo.setLikes(imageInfoDB.getLikes());
        imageInfo.setViews(imageInfoDB.getViews());
        imageInfo.setHeight(imageInfoDB.getHeight());
        imageInfo.setWidth(imageInfoDB.getWidth());
        return imageInfo;
    }
}

File: app/src/main/java/com/mss/imagesearcher/repositories/network/NetMapper.kt
package com.mss.imagesearcher.repositories.network

import com.mss.imagesearcher.domain.models.ImageInfo
import com.mss.imagesearcher.repositories.network.models.ImageInfoNet
import java.util.*

object NetMapper {

    fun mapFromNet(imagesNet: List<ImageInfoNet>): List<ImageInfo> {
        val images = ArrayList<ImageInfo>()
        for (imageNet in imagesNet) {
            images.add(mapFromNet(imageNet))
        }
        return images
    }

    private fun mapFromNet(imageInfoDB: ImageInfoNet) = ImageInfo(
            imageInfoDB.id,
            imageInfoDB.largeImageURL,
            imageInfoDB.previewURL,
            imageInfoDB.webFormatURL,
            imageInfoDB.height,
            imageInfoDB.width,
            imageInfoDB.type,
            imageInfoDB.tags,
            imageInfoDB.pageUrl,
            imageInfoDB.likes,
            imageInfoDB.views,
            imageInfoDB.comments
    )
}
